client/src/routes/Session/hooks/useUpdateSessionState.ts

```diff
@@ -0,0 +1,83 @@
+import {useCallback, useMemo} from 'react';
+import {ExerciseSlide} from '../../../../../shared/src/types/Content';
+import {SessionState, Session} from '../../../../../shared/src/types/Session';
+import * as sessionApi from '../../../lib/sessions/api/session';
+
+const useUpdateSessionState = (sessionId: Session['id'] | undefined) => {
+  const startSession = useCallback(async () => {
+    if (sessionId) {
+      sessionApi.updateSessionState(sessionId, {
+        started: true,
+      });
+    }
+  }, [sessionId]);
+
+  const endSession = useCallback(async () => {
+    if (sessionId) {
+      sessionApi.updateSessionState(sessionId, {
+        ended: true,
+      });
+    }
+  }, [sessionId]);
+
+  const navigateToIndex = useCallback(
+    async ({
+      index,
+      content,
+    }: {
+      index: SessionState['index'];
+      content: ExerciseSlide[];
+    }) => {
+      if (!sessionId || index < 0 || index > content.length - 1) {
+        return;
+      }
+
+      const completed = index === content.length - 1 ? true : undefined;
+
+      return sessionApi.updateSessionState(sessionId, {
+        index,
+        playing: false,
+        completed,
+      });
+    },
+    [sessionId],
+  );
+
+  const setSpotlightParticipant = useCallback(
+    async (dailySpotlightId: SessionState['dailySpotlightId']) => {
+      if (sessionId) {
+        return sessionApi.updateSessionState(sessionId, {
+          dailySpotlightId,
+        });
+      }
+    },
+    [sessionId],
+  );
+
+  const setPlaying = useCallback(
+    async (playing: SessionState['playing']) => {
+      if (sessionId) {
+        return sessionApi.updateSessionState(sessionId, {playing});
+      }
+    },
+    [sessionId],
+  );
+
+  return useMemo(
```

Comment:
Don't think we need memo here, if we always use it like `const {setPlaying} = useUpdateSessionState()` but maybe we don't always do? If we somewhere use it like `const updateSessionState = useUpdateSessionState()` memo is needed